src/vsphere_cpi/spec/integration/cluster_different_datastores_spec.rb

```diff
@@ -33,72 +33,123 @@
       'cpu' => 1,
     }
   end
-  let(:first_datastore_cpi) do
-    options = cpi_options(
-      datacenters: [{
-        datastore_pattern: @datastore_pattern,
-        persistent_datastore_pattern: @datastore_pattern,
-      }],
-    )
-    VSphereCloud::Cloud.new(options)
-  end
-  let(:second_datastore_cpi) do
-    options = cpi_options(
-      datacenters: [{
-        datastore_pattern: @second_datastore,
-        persistent_datastore_pattern: @second_datastore,
-      }],
-    )
-    VSphereCloud::Cloud.new(options)
+
+  context 'randomization for datastore selection is set to zero' do
+    # Mocking out stable random to a constant 1 to ensure algorithm scores
+    # datastores in descending order of free space.
+    module VSphereCloud::SelectionPipeline::StableRandom
+      def self.[](object)
+        1
+      end
+    end
+    let(:all_datastore_cpi) do
+      options = cpi_options(
+        datacenters: [{
+          datastore_pattern: '.*',
+          persistent_datastore_pattern: '.*',
+        }],
+      )
+      VSphereCloud::Cloud.new(options)
+    end
+
+    it 'should pick the datastore with most amount of free space' do
+      cluster_resource = all_datastore_cpi.datacenter.find_cluster(@cluster_name)
+      max_free_space_datastore = cluster_resource.mob.datastore.sort_by do |ds|
```

Comment:
Can this be not a specific datastore ? If according to our env setup if you expect it is datastore-x then lets use that instead of dynamic value for datastore

Reply:
This will be `nsf0-1` but in some cases if there is lot of stemcell replication or repeated BATS test, `vsandatstore` might trump `nfs0-1` in free space. Therefore, it is not possible to keep it static.